pkg/catalog/outbound_traffic_policies.go

```diff
@@ -21,14 +26,14 @@ func (mc *MeshCatalog) ListOutboundTrafficPolicies(downstreamIdentity identity.S
 	downstreamServiceAccount := downstreamIdentity.ToK8sServiceAccount()
 	if mc.configurator.IsPermissiveTrafficPolicyMode() {
 		var outboundPolicies []*trafficpolicy.OutboundTrafficPolicy
-		mergedPolicies := trafficpolicy.MergeOutboundPolicies(DisallowPartialHostnamesMatch, outboundPolicies, mc.buildOutboundPermissiveModePolicies(downstreamServiceAccount.Namespace)...)
+		mergedPolicies := trafficpolicy.MergeOutboundPolicies(DisallowPartialHostnamesMatch, trafficpolicy.RetryPolicy{}, outboundPolicies, mc.buildOutboundPermissiveModePolicies(downstreamServiceAccount.Namespace)...)
```

Comment:
Now that the retry policy is a part of the route, instead of passing empty struct like this, fill in the retry policy in the OutboundPolicy struct and have the merge function resolve merges. 

Is there a need to pass empty structs around?

Reply:
@shashankram do you mean changing it to something like below
```
mergedPolicies := trafficpolicy.MergeOutboundPolicies(DisallowPartialHostnamesMatch, outboundPolicies[0].Routes[0].RetryPolicy, outboundPolicies, mc.buildOutboundPermissiveModePolicies(downstreamServiceAccount.Namespace)...)
```

Reply:
I mean't keeping it as it was before, but updating the trafficpolicy struct with the Retry policy before the merge. We can have the Merge function deal with merging however it deems correct. Would that work?